File: examples/src/Slides/Hyperlinks/MacroHyperLink.cs
using System
using System.Collections.Generic
using System.IO
using System.Linq
using System.text
using System.Threading.Tasks
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using Aspose.slides.Export

/*
This code example demonstrates how the SetMacroHyperlinkClick method is used to set a macro hyperlink click on a shape:
*/

namespace Aspose.slides.Examples.CSharp.slides.Hyperlinks
{
    public class MacroHyperlink
    {
        public static void Run()
        {
            macroName = "TestMacro"
            with slides.Presentation() as presentation:
            {
                shape = presentation.slides[0].shapes.add_auto_shape(ShapeType.BlankButton, 20, 20, 80, 30)
                shape.HyperlinkManager.SetMacroHyperlinkClick(macroName)

                print("External URL is {0}", shape.HyperlinkClick.ExternalUrl)
                print("Shape action type is {0}", shape.HyperlinkClick.ActionType)
            }
        }
    }
}

File: examples/src/slides/hyperlinks/macro_hyperlink.py
import aspose.slides as slides


def macro_hyperlink():
    """This code example demonstrates how the set_macro_hyperlink_click method is used
    to set a macro hyperlink click on a shape"""

    macro_name = "TestMacro"
    with slides.Presentation() as presentation:
        shape = presentation.slides[0].shapes.add_auto_shape(slides.ShapeType.BLANK_BUTTON, 20, 20, 80, 30)
        shape.hyperlink_manager.set_macro_hyperlink_click(macro_name)

        print("External URL is {0}".format(shape.hyperlink_click.external_url))
        print("Shape action type is {0}".format(shape.hyperlink_click.action_type))
